pkg/resultstore/syncset.go

```diff
@@ -0,0 +1,260 @@
+package resultstore
+
+import (
+	"sync"
+
+	"gopkg.in/typ.v3/pkg/sets"
+)
+
+// NewSyncSet creates a new empty sync set.
+func NewSyncSet[T comparable]() SyncSet[T] {
+	return NewSyncSetFromSet(make(sets.Set[T]))
+}
+
+// NewSyncSetFromSet creates a new sync set that is a clone of the provided set.
+func NewSyncSetFromSet[T comparable](set sets.Set[T]) SyncSet[T] {
+	return syncSet[T]{
+		s:  set.Clone(),
+		mu: &sync.RWMutex{},
+	}
+}
+
+// SyncSet is an interface for thread-safe wrappers of a set.
+type SyncSet[T comparable] interface {
+	// String converts this set to its string representation.
```

Comment:
This is a lot of code. Try removing all methods except the ones we really need.

Also, no need to export this type :) It's just a small utility wrapper, and having an exported `SyncSet` doesn't really belong in this package, and we don't really need to maintain a fully-featured wrapper anyway